subsys/mpsl/mpsl_init.c

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <init.h>
+#include <irq.h>
+#include <kernel.h>
+#include <mpsl.h>
+
+void m_assert_handler(const char *const file, const u32_t line)
+{
+	k_oops();
+}
+
+static int mpsl_lib_init(struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	int err = 0;
+	mpsl_clock_lf_cfg_t clock_cfg;
+
+#ifdef CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC
+	clock_cfg.source = MPSL_CLOCK_LF_SRC_RC;
+#elif CONFIG_CLOCK_CONTROL_NRF_K32SRC_XTAL
+	clock_cfg.source = MPSL_CLOCK_LF_SRC_XTAL;
+#elif CLOCK_CONTROL_NRF_K32SRC_SYNTH
+	clock_cfg.source = MPSL_CLOCK_LF_SRC_SYNTH;
+#else
+#error "Clock source is not defined"
```

Comment:
What about `_EXT_LOW_SWING` and `_EXT_FULL_SWING` options, are they supposed to be not supported?

Reply:
Right, we don't support the other sources. Do you think the error message should be different?

Reply:
I think so. The current one may be a bit confusing.